src/scores/continuous/consistent_impl.py

```diff
@@ -0,0 +1,262 @@
+"""
+Implementation of scoring functions that are consistent for
+single-valued forecasts targeting quantiles, expectiles or Huber functionals.
+"""
+
+from typing import Callable, Optional
+
+import xarray as xr
+
+from scores.functions import apply_weights
+from scores.typing import FlexibleDimensionTypes
+from scores.utils import gather_dimensions
+
+
+def consistent_expectile_score(
+    fcst: xr.DataArray,
+    obs: xr.DataArray,
+    *,  # Force keywords arguments to be keyword-only
+    alpha: float,
+    phi: Callable[[xr.DataArray], xr.DataArray],
+    phi_prime: Callable[[xr.DataArray], xr.DataArray],
+    reduce_dims: Optional[FlexibleDimensionTypes] = None,
+    preserve_dims: Optional[FlexibleDimensionTypes] = None,
+    weights: Optional[xr.DataArray] = None,
+) -> xr.DataArray:
+    """
+    Returns the score using a scoring function that is consistent for the
+    alpha-expectile functional, based on a supplied convex function phi.
+    See Geniting (2011), or Equation (10) from Taggart (2021).
+
+    Args:
+        fcst: array of forecast values.
+        obs: array of corresponding observation values.
+        alpha: expectile level. Must be strictly between 0 and 1.
+        phi: a convex function on the real numbers, accepting a single array like argument.
+        phi_prime: a subderivative of `phi`, accepting a single array like argument.
+        reduce_dims: Optionally specify which dimensions to reduce when
+            calculating the consistent expectile score. All other dimensions will be preserved. As a
+            special case, 'all' will allow all dimensions to be reduced. Only one
+            of `reduce_dims` and `preserve_dims` can be supplied. The default behaviour
+            if neither are supplied is to reduce all dims.
+        preserve_dims: Optionally specify which dimensions to preserve when calculating
+            the consistent quantile score. All other dimensions will be reduced. As a special case, 'all'
+            will allow all dimensions to be preserved. In this case, the result will be in
+            the same shape/dimensionality as the forecast, and the errors will be the consistent quantile
+            score at each point (i.e. single-value comparison against observed), and the
+            forecast and observed dimensions must match precisely. Only one of `reduce_dims`
+            and `preserve_dims` can be supplied. The default behaviour if neither are supplied
+            is to reduce all dims.
+        weights: Optionally provide an array for weighted averaging (e.g. by area, by latitude,
+            by population, custom)
+
+    Returns:
+        array of (mean) scores that is consistent for alpha-expectile functional,
```

Comment:
It would be useful to know the range of the score - i.e. is it between 0 and 1, is 0 perfect or worst, etc, just to give some hints. Or, alternatively, some comment like "this is a specialised score - please review this tutorial for an explanation of the arguments and returned values"